internal/expressions/function/functions.go

```diff
@@ -609,6 +632,49 @@ func getStatus(
 	}
 }
 
+// semverDiff compares two semantic version strings and returns the difference
+// level between them.
+//
+// It returns one of: "MAJOR", "MINOR", "PATCH", "METADATA", "NONE", or
+// "INCOMPARABLE" if either version is invalid.
+func semverDiff(a ...any) (any, error) {
+	if len(a) != 2 {
+		return nil, fmt.Errorf("expected 2 arguments, got %d", len(a))
+	}
+
+	ver1Str, ok := a[0].(string)
+	if !ok {
+		return nil, fmt.Errorf("first argument must be string, got %T", a[0])
+	}
+
+	ver2Str, ok := a[1].(string)
+	if !ok {
+		return nil, fmt.Errorf("second argument must be string, got %T", a[1])
+	}
+
+	ver1, err := semver.NewVersion(ver1Str)
+	if err != nil {
+		return "INCOMPARABLE", nil
+	}
+	ver2, err := semver.NewVersion(ver2Str)
+	if err != nil {
+		return "INCOMPARABLE", nil
+	}
+	if ver1.Major() != ver2.Major() {
+		return "MAJOR", nil
+	}
+	if ver1.Minor() != ver2.Minor() {
+		return "MINOR", nil
+	}
+	if ver1.Patch() != ver2.Patch() {
+		return "PATCH", nil
+	}
+	if ver1.Metadata() != ver2.Metadata() {
+		return "METADATA", nil
+	}
+	return "NONE", nil
```

Comment:
- Do these have to be all caps?
- I wonder if, instead of `NONE` (and/or `INCOMPARABLE`), it wouldn't be better to return an empty string? This would make certain comparisons easier if you only care about a change.

Reply:
> distinctly different outcomes

That a user would often wish to handle differently.

Reply:
I quite like the idea of a second function that distills the results to a bool. I still worry that None and Incomparable mutually collapsing into `false` make no real diff indistinguishable from just bad input.

I think the _generalized_ problem here is expr-lang doesn't really have a convenient way of dealing with errors (that I yet know of).

But I think something returning a boolean is a good idea.

It might be good to take it a step farther with an optional third param...

`semverHasDiff(v1, v2, 'Minor')`, for example, could answer whether the diff is minor or bigger and would return false for patch or metadata diffs.

Reply:
The only problem with the boolean is there's no way to account for failures.

I still like the idea and think we could revisit `semverHasDiffs()` in a follow-up.